File: src/test/java/common/factory/WebDriverWaitFactory.java
package common.factory;

import io.appium.java_client.AppiumDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

import static common.SharedConstants.TIMEOUT_MILLISECONDS;

public class WebDriverWaitFactory {

    public static WebDriverWait create(final AppiumDriver driver) {
        return create(driver, TIMEOUT_MILLISECONDS);
    }

    public static WebDriverWait create(final AppiumDriver driver, final long timeoutMilliseconds) {
        return new WebDriverWait(driver, timeoutMilliseconds / 1000);
    }
}

File: src/test/java/common/factory/WebDriverWaitFactory.kt
package common.factory

import common.SharedConstants
import io.appium.java_client.AppiumDriver
import org.openqa.selenium.support.ui.WebDriverWait
import java.time.Duration

object WebDriverWaitFactory {

    @JvmOverloads
    fun create(
        driver: AppiumDriver,
        timeoutMilliseconds: Long = SharedConstants.TIMEOUT_MILLISECONDS
    ): WebDriverWait {
        return WebDriverWait(driver, Duration.ofMillis(timeoutMilliseconds))
    }

}